cloud/terraform/base_config_builder.py

```diff
@@ -38,9 +38,18 @@ def create_resource_name(self, resource_names_combination, separator='-') -> str
         :param separator: String to be used as a separator of the items passed in resource_names_combination
         :return: String composed of a prefix, resource names combination and a suffix with random numbers.
         """
-        combinations = [self.resource_name_prefix]
-        combinations.extend(resource_names_combination)
-        combinations.append(self.get_random_numbers())
+        tf_resource_name_max_chars = 63
+        random_num = self.get_random_numbers()
+
+        combined_name = separator.join(resource_names_combination).replace(f'{self.resource_name_prefix}-', '')
```

Comment:
Is the prefix ever part of `resource_names_combination` ?  Removing it feels a bit redundant if it isn't part of the input:

Reply:
In the cloud-specific config builders we use some dependent resource names and they do have the prefix already. This is to avoid having the prefix set multiple times, which also increases the length and adds extra prefixes where it shouldn't. That's why we remove it in case it was already added previously.

Reply:
:man_facepalming: 

This really needs to be documented in the source code because it isn't obvious why it needs to work like that.